src/metabase/api/field.clj

```diff
@@ -419,13 +419,14 @@
     value        :- [:maybe ms/NonBlankString]
     maybe-limit  :- [:maybe ms/PositiveInt]]
    (try
-     (let [field   (follow-fks field)
-           limit   (or maybe-limit default-max-field-search-limit)
-           results (qp/process-query (search-values-query field search-field value limit))]
-       (get-in results [:data :rows]))
-     (catch Throwable e
-       (log/error e (trs "Error searching field values"))
-       nil))))
+    (let [field        (follow-fks field)
+          search-field (follow-fks search-field)
```

Comment:
the fix is just wrapping `search-field` with `follow-fks`

Reply:
I know this is confused, and I don't think I explained the above problem well, feel free to ping me, I'll give you a walk through. Thank you!